src/devices/SoftPwm/SoftwarePwmChannel.cs

```diff
@@ -197,8 +209,11 @@ protected override void Dispose(bool disposing)
             _runThread = false;
             _runningThread?.Join();
             _runningThread = null;
-            _controller?.Dispose();
-            _controller = null;
+            if (_shouldDispose)
+            {
+                _controller?.Dispose();
+                _controller = null;
+            }
```

Comment:
nit: enter after `}` and before `if` statement (here and below)

Reply:
Done, but should add StyleCop for that (I guess that's why you created #816).